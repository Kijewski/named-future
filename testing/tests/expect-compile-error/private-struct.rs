use futures_lite::future::block_on;

mod inner {
    #[named_future::named_future(type = Answer)]
    pub async fn answer() -> usize {
        42
    }
}

fn main() {
    let future: inner::Answer = inner::answer();
    assert_eq!(42, block_on(future));
}
