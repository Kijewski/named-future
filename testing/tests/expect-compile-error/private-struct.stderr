error[E0603]: struct `Answer` is private
  --> tests/expect-compile-error/private-struct.rs:11:24
   |
11 |     let future: inner::Answer = inner::answer();
   |                        ^^^^^^ private struct
   |
note: the struct `Answer` is defined here
  --> tests/expect-compile-error/private-struct.rs:5:18
   |
5  |     pub async fn answer() -> usize {
   |                  ^^^^^^
